src/eval.rs

```diff
@@ -46,13 +48,33 @@ impl<T: AsRef<str>> EvalString<T> {
         }
     }
 
+    fn calc_evaluated_length(&self, envs: &[&dyn Env]) -> usize {
+        self.0
+            .iter()
+            .map(|part| match part {
+                EvalPart::Literal(s) => s.as_ref().len(),
+                EvalPart::VarRef(v) => {
+                    let mut result = 0;
+                    for (i, env) in envs.iter().enumerate() {
+                        if let Some(v) = env.get_var(v.as_ref()) {
+                            result = v.calc_evaluated_length(&envs[i + 1..]);
```

Comment:
This is better as `return v.calc_eval...` with a `return 0` at the end of the loop

Reply:
Good point, done.